manager/integration/tests/test_basic.py

```diff
@@ -733,7 +729,71 @@ def test_backup_volume_list():  # NOQA
     11. delete backup volumes(1 & 2)
     12. cleanup
     """
-    pass
+    backupstore_cleanup(client)
+
+    # create 2 volumes.
+    volume1_name, volume2_name = generate_volume_name(), generate_volume_name()
+
+    volume1 = create_and_check_volume(client, volume1_name)
+    volume2 = create_and_check_volume(client, volume2_name)
+
+    host_id = get_self_host_id()
+    volume1 = volume1.attach(hostId=host_id)
+    volume1 = common.wait_for_volume_healthy(client, volume1_name)
+    volume2 = volume2.attach(hostId=host_id)
+    volume2 = common.wait_for_volume_healthy(client, volume2_name)
+
+    bv1, backup1, snap1, _ = create_backup(client, volume1_name)
+    bv2, backup2, snap2, _ = create_backup(client, volume2_name)
+
+    def verify_no_err():
+        '''
+        request a backup list
+        verify backup list contains no error messages for volume(1,2)
+        verify backup list contains backup(1) for volume(1,2)
+        '''
+        for _ in range(RETRY_COUNTS):
+            verified_bvs = set()
+            backup_volume_list = client.list_backupVolume()
+            for bv in backup_volume_list:
+                if bv.name in (volume1_name, volume2_name):
+                    assert not bv['messages']
+                    for b in bv.backupList().data:
+                        if bv.name == volume1_name \
+                                and b.name == backup1.name \
+                                or bv.name == volume2_name \
+                                and b.name == backup2.name:
+                            verified_bvs.add(bv.name)
+            if len(verified_bvs) == 2:
+                break
+            time.sleep(RETRY_INTERVAL)
+        assert len(verified_bvs) == 2
+
+    verify_no_err()
+
+    # place a bad named file into the backups folder of volume(1)
+    prefix = minio_get_backup_volume_prefix(volume1_name) + "/backups"
+    backupstore_create_file(client,
+                            core_api,
+                            prefix,
+                            "backup_1234@failure.cfg")
+
+    verify_no_err()
+
+    backupstore_delete_file(client,
+                            core_api,
+                            prefix,
+                            "backup_1234@failure.cfg")
+
+    bv1 = client.by_id_backupVolume(volume1_name)
```

Comment:
duplicate cleanup since the test will call `backupstore_cleanup` at the end.
Same below.

Reply:
fixed